On-Premise/JavaScript/CustomSamples/CreateNewCategorySample/README.md

```diff
@@ -0,0 +1,41 @@
+# How to Add a New Category Button While Saving a Report on the Server using Bold Reports Designer
+
+This project utilizes JavaScript to demonstrate the process of adding a new category button while saving a report using a sample JavaScript application.
+
+## Requirements to run the sample
+
+To run the sample, ensure you have the following requirements:
+
+* [Visual Studio Code](https://code.visualstudio.com/download)
+
+## Using the Reporting Samples
+
+Follow these steps to utilize the reporting samples:
+
+1. Open the html file `CreateCategory.html` in Visual Studio Code.
+
+2. Press `F5` or click the `Run` button in Visual Studio Code to launch the application.
+
+## Why This Sample?
+
+The provided code snippet illustrates how to customize the Bold Reports server integration within a JavaScript application. By initializing the Bold Reports Designer with the necessary server URLs and authorization tokens, users can dynamically add a **New Category** button to the report designer's UI. This customization ensures that the button is rendered only once during the initial save action, streamlining the report-saving process.
+
+## Implementation Details
+
+1. **Configure Application:** Initialize the Bold Reports Designer with relevant server details.
+2. **Add New Category Button:** Define the [create](https://help.boldreports.com/embedded-reporting/javascript-reporting/report-designer/api-reference/events/#create) API event function to add the **New Category** button to the UI dynamically.
+![CategoryButton.png](https://support.boldreports.com/kb/agent/attachment/inline?token=eyJhbGciOiJodHRwOi8vd3d3LnczLm9yZy8yMDAxLzA0L3htbGRzaWctbW9yZSNobWFjLXNoYTI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjIzODIyIiwib3JnaWQiOiIzIiwiaXNzIjoic3VwcG9ydC5ib2xkcmVwb3J0cy5jb20ifQ.l_QK7rgTnIofHNcqyMJAEYngrCvO3TaC3oZ7MxPOkB0)
```

Comment:
No need to blur it. its just file name only


![image](https://github.com/boldreports/samples/assets/74242362/3cc7fbda-4d0e-46db-b6ac-042799dd5982)